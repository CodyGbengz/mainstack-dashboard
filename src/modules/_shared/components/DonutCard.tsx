import styled from 'styled-components';

import { DoughnutChart } from './Charts/Doughnut';
import { DonutSummary } from './DonutSummary';
import Typography from './Typography';
import { Column } from './Column';
import { Row } from './Row';

export type DonutCardProps = {
  title: string;
  chartData: any;
};

export const DonutCard: React.FC<DonutCardProps> = ({ title, chartData }) => (
  <DonutCardContainer>
    <TopCardRow>
      <Row>
        <Typography fontWeight='bold'>{title}</Typography>
        <Typography
          fontWeight='thin'
          fontSize='0.875rem'
          color='primary'
        >
          View full reports
        </Typography>
      </Row>
    </TopCardRow>
    <Row>
      <Column>
        <DonutSummary chartData={chartData} />
      </Column>
      <Column>
        <DoughnutChartWrapper>
          <DoughnutChart chartData={chartData} />
        </DoughnutChartWrapper>
      </Column>
    </Row>
  </DonutCardContainer>
);

const DonutCardContainer = styled.div`
  border: 1px solid #eff1f6;
  border-radius: 0.75rem;
  height: 16.375rem;
  padding: 1.5rem;
  width: 31.25rem;
`;

const TopCardRow = styled.div`
  margin: 0 0 3rem 0;
`;

const DoughnutChartWrapper = styled.div`
  height: 10rem;
  width: 10rem;
`;
