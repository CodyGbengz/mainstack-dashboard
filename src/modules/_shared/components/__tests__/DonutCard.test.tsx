import { render, screen } from '@testing-library/react';
import { DonutCard, DonutCardProps } from '../DonutCard';

describe('COMPONENT TEST: DonutCard', () => {
  const chartTestData = [
    {
      country: 'Nigeria',
      count: 68,
      percent: 34,
    },
    {
      country: 'Germany',
      count: 37,
      percent: 19,
    },
  ];

  const setup = (props: Partial<DonutCardProps> = {}) =>
    render(
      <DonutCard
        title='My test card'
        chartData={chartTestData}
        {...props}
      />
    );

  test('Should match snapshot', () => {
    const { container } = setup();
    expect(container.parentElement).toMatchSnapshot();
  });

  test('should render title -My test card', () => {
    setup();
    expect(
      screen.getByText('My test card', { exact: false })
    ).toBeInTheDocument();
  });

  test('should render correctly with content', () => {
    setup();
    expect(screen.getByText('Nigeria', { exact: false })).toBeInTheDocument();
    expect(screen.getByText('Germany', { exact: false })).toBeInTheDocument();
    expect(screen.getByText('34', { exact: false })).toBeInTheDocument();
    expect(screen.getByText('19', { exact: false })).toBeInTheDocument();
  });
});
