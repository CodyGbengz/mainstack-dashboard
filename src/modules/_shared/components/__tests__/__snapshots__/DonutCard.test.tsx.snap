// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`COMPONENT TEST: DonutCard > Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-eCstFB jDGZCq"
    >
      <div
        class="sc-jSgtCu BVkWv"
      >
        <div
          class="sc-hKgJyM riRPJ"
        >
          <p
            class="sc-bdfCkl inwahD"
            font-weight="bold"
          >
            My test card
          </p>
          <p
            class="sc-bdfCkl hJmnRW"
            color="#FF5403"
            font-size="0.875rem"
            font-weight="thin"
          >
            View full reports
          </p>
        </div>
      </div>
      <div
        class="sc-hKgJyM riRPJ"
      >
        <div
          class="sc-dlfmnT cZSXsv"
        >
          <div
            style="display: flex; gap: 0.5rem; align-items: center;"
          >
            <span
              class="fi fi-ng"
            />
            <p
              class="sc-bdfCkl fKJepU"
              font-size="0.875rem"
              font-weight="thin"
            >
              Nigeria
               
              34
              %
            </p>
            <div
              class="sc-gsTChe NqgVI"
              color="rgba(89, 158, 234, 1)"
            />
          </div>
          <div
            style="display: flex; gap: 0.5rem; align-items: center;"
          >
            <span
              class="fi fi-de"
            />
            <p
              class="sc-bdfCkl fKJepU"
              font-size="0.875rem"
              font-weight="thin"
            >
              Germany
               
              19
              %
            </p>
            <div
              class="sc-gsTChe jllppm"
              color="rgba(132, 79, 246, 1)"
            />
          </div>
        </div>
        <div
          class="sc-dlfmnT cZSXsv"
        >
          <div
            class="sc-gKsdJj gZkwhB"
          >
            <canvas
              height="150"
              role="img"
              width="300"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
